codebase2/codebase/U/Codebase/ShortHash.hs

```diff
@@ -20,3 +21,7 @@ data ShortHash
   deriving (Eq, Ord, Show)
 
 data ShortBranchHash = ShortBranchHash {toText :: Text} deriving (Eq, Ord, Show)
+
+shortenTo :: Int -> ShortHash -> ShortHash
+shortenTo _ b@(Builtin _) = b
+shortenTo i s@ShortHash {..} = s {prefix = Text.take i prefix}
```

Comment:
Reshortening a `ShortHash` seems weird, what if `i` is bigger than the current length? Maybe it would make sense to hold onto a `Reference`/`Referent` until the last minute?

Reply:
It doesn't have to be in this PR, but wdyt

Reply:
I agree this seems strange, I think the length should probably be specified at construction time, but this is currently just a copy of `take` from the V1 Shorthash, so if we want to change the way this is done we should use that strategy consistently.